src/sutta_publisher/edition_parsers/base.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC
+from typing import Type
+
+from sutta_publisher.shared.value_objects.edition import EditionResult, EditionType
+from sutta_publisher.shared.value_objects.edition_config import EditionConfig
+from sutta_publisher.shared.value_objects.edition_data import EditionData
+
+log = logging.getLogger(__name__)
+
+
+class EditionParser(ABC):
+    config: EditionConfig
+    raw_data: EditionData
+    edition_type: EditionType
+
+    def __init__(self, config: EditionConfig, data: EditionData) -> None:
+        self.raw_data = data
+        self.config = config
```

Comment:
I am confused, are config and raw_data supposed to be both class attributes and instance attributes?
If so, won't there be any scope conflicts / side effects we should watch out for?